common/test/unit/com/thoughtworks/go/domain/materials/git/GitCommandTest.java

```diff
@@ -358,15 +358,20 @@ public void shouldRetrieveZeroSubmoduleUrlsIfTheyAreNotConfigured() throws Excep
 
     @Test public void shouldCheckIfRemoteRepoExists() throws Exception {
```

Comment:
Write more test cases for substring match to check if the branch names are exact